alacritty/src/renderer/mod.rs

```diff
@@ -462,11 +465,25 @@ impl Batch {
         Batch { tex: 0, instances: Vec::with_capacity(BATCH_MAX) }
     }
 
-    pub fn add_item(&mut self, cell: RenderableCell, glyph: &Glyph) {
+    pub fn add_item(&mut self, mut cell: RenderableCell, glyph: &Glyph) {
         if self.is_empty() {
             self.tex = glyph.tex_id;
         }
 
+        if glyph.colored {
+            // XXX Temporary workaround to prevent emojis being rendered with a wrong colors on, at
+            // least, dark backgrounds. To resolve this issue properly we should have a separate
+            // rendering pass for colored glyphs. For more info see #1864.
+            //
+            // The 255 is not a color, it'll be a part of forming an alpha mask later on. So a
+            // proper color of emojis could be achieved by setting fg to #ffffff and bg
+            // to #000000, however we can't change bg here, because it actually "changes
+            // bg" of a cell.
```

Comment:
There's a lot of complexity to this issue, so I don't feel like going into detail here brings us much benefit and might confuse people rather than help as documentation.

I think just referring to #1864 should be sufficient, then we can document things there and people can read up on the complete extensive documentation, rather than making assumptions based on incomplete data.